docs/backend/subscribers.md

```diff
@@ -11,8 +11,268 @@ myst:
+
+```xml
+<subscriber
+    for=".interfaces.IMyObject
+         .interfaces.IMyEvent"
+    handler=".content.MyObject.myEventHandler"
+    />
+```
+
+The following example shows how to subscribe a content type to the life cycle event.
+
+```xml
+<subscriber
+    zcml:condition="installed zope.lifecycleevent"
+    for=".interfaces.ISitsPatient
+         zope.lifecycleevent.IObjectModifiedEvent"
+    handler=".content.SitsPatient.objectModified"
+    />
+```
+
+
+### Subscribe to an event using Python
+
+The following example shows how subscription is valid through the process life cycle.
+In unit tests, it is important to clear test event handlers between the test steps.
+
+```python
+import zope.component
+
+def my_event_handler(context, event):
+    """
+    @param context: Zope object for which the event was fired. Usually this is a Plone content object.
+
+    @param event: Subclass of event.
+    """
+    pass
+
+gsm = zope.component.getGlobalSiteManager()
+gsm.registerHandler(my_event_handler, (IMyObject,IMyEvent))
+```
+
+
+## Fire an event
+
+Use `zope.event.notify()` to fire event objects to their subscribers.
+
+The following code shows how to fire an event in unit tests.
+
+```python
+import zope.event
+from plone.postpublicationhook.event import AfterPublicationEvent
+
+event = AfterPublicationEvent(self.portal, self.portal.REQUEST)
+zope.event.notify(event)
+```
+
+
+(subscribers-event-types-label)=
+
+## Event types
+
+Plone has the following types of events.
+
+
+### Creation events
+
+`zope.lifecycleevent.IObjectCreatedEvent` is fired for all Zope-ish objects when they are created, or copied via `IObjectCopiedEvent`.
+They don't have to be content objects.
+
+### Modified events
+
+`zope.lifecycleevent.IObjectModifiedEvent` is called for creation stage events as well, unlike the previous event type.
+
+### Delete events
+
+Delete events can be fired several times for the same object.
+Some delete event transactions are rolled back.
+
+### Copy events
+
+`zope.lifecycleevent.IObjectCopiedEvent` is triggered when an object is copied.
+It will also fire `IObjectCreatedEvent` event code.
+
+### Workflow events
+
+`Products.DCWorkflow.interfaces.IBeforeTransitionEvent` is triggered before a workflow transition is executed.
+
+`Products.DCWorkflow.interfaces.IAfterTransitionEvent` is triggered after a workflow transition has been executed.
+
+The DCWorkflow events are low-level events that can tell you a lot about the previous and current states.
+
+`Products.CMFCore.interfaces.IActionSucceededEvent` is a higher level event that is more commonly used to react after a workflow action has completed.
+
+### Zope startup events
+
+`zope.processlifetime.IProcessStarting` is triggered after the component registry has been loaded and Zope is starting up.
+
+`zope.processlifetime.IDatabaseOpened` is triggered after the main ZODB database has been opened.
+
+
+(subscribers-event-handlers)=
```

Comment:
I stopped my review here. The foregoing content appears to be a rehash of the preceding content. I am not qualified to decide what should be kept, edited, or tossed out.

Reply:
AFAIKS the content below is already described above and can be removed.